custom_components/pyscript/__init__.py

```diff
@@ -217,9 +217,16 @@ async def state_changed(event):
             "value": new_val,
             "old_value": old_val,
             "context": event.context,
-            "old_state": event.data.get('old_state'),
-            "new_state": event.data.get('new_state'),
         }
+
+        if new_val is not None:
+            for attribute in event.data["new_state"].attributes:
+                new_vars[f"{var_name}.{attribute}"] = event.data["new_state"].attributes[attribute]
+
+        if old_val is not None:
+            for attribute in event.data["old_state"].attributes:
+                new_vars[f"{var_name}.{attribute}.old"] = event.data["old_state"].attributes[attribute]
+
```

Comment:
These lines shouldn't be necessary now, since the `value` and `old_value` contain the full states.